src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopySendTemporalDataTypesAsStringTest.java

```diff
@@ -0,0 +1,225 @@
+ * sent as MONEY / SMALLMONEY datatypes instead of DECIMAL after these changes.
+ * 
+ * Note that with this connection property set to FALSE, the driver will only accept the default string literal format
+ * of each temporal datatype, for example:
+ * 
+ * DATE: YYYY-MM-DD
+ * DATETIME: YYYY-MM-DD hh:mm:ss[.nnn]
+ * DATETIME2: YYYY-MM-DD hh:mm:ss[.nnnnnnn]
+ * DATETIMEOFFSET: YYYY-MM-DD hh:mm:ss[.nnnnnnn] [{+|-}hh:mm]
+ * SMALLDATETIME:YYYY-MM-DD hh:mm:ss
+ * TIME: hh:mm:ss[.nnnnnnn]
+ * 
+ */
+@RunWith(JUnitPlatform.class)
+public class BulkCopySendTemporalDataTypesAsStringTest extends AbstractTest {
+    static String inputFile = "BulkCopyCSVSendTemporalDataTypesAsStringForBulkCopy.csv";
+    static String encoding = "UTF-8";
+    static String delimiter = ",";
+
+    static String destTableName = AbstractSQLGenerator
+            .escapeIdentifier(RandomUtil.getIdentifier("sendTemporalDataTypesAsStringForBulkCopyDestTable"));
+    static String destTableName2 = AbstractSQLGenerator
+            .escapeIdentifier(RandomUtil.getIdentifier("sendTemporalDataTypesAsStringForBulkCopyDestTable2"));
+    static String filePath = null;
+
+    /**
+     * Test basic case with sendTemporalDataTypesAsStringForBulkCopy connection property.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSendTemporalDataTypesAsStringForBulkCopy() throws SQLException {
+        beforeEachSetup();
+        try (Connection conn = PrepUtil
+                .getConnection(connectionString + ";sendTemporalDataTypesAsStringForBulkCopy=false")) {
+            SQLServerBulkCSVFileRecord fileRecord = new SQLServerBulkCSVFileRecord(filePath + inputFile, encoding,
+                    delimiter, true);
+
+            testBulkCopyCSV(conn, fileRecord);
+        }
+    }
+
+    /**
+     * Test basic case with sendTemporalDataTypesAsStringForBulkCopy connection property, using a resultset.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSendTemporalDataTypesAsStringForBulkCopyRS() throws SQLException {
+        beforeEachSetup();
+        try (Connection conn = PrepUtil
+                .getConnection(connectionString + ";sendTemporalDataTypesAsStringForBulkCopy=false")) {
+            SQLServerBulkCSVFileRecord fileRecord = new SQLServerBulkCSVFileRecord(filePath + inputFile, encoding,
+                    delimiter, true);
+
+            testBulkCopyResultSet(conn, fileRecord);
+        }
+    }
+
+    /**
+     * Test basic case with sendTemporalDataTypesAsStringForBulkCopy connection property, using a data source.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSendTemporalDataTypesAsStringForBulkCopyDS() throws SQLException {
+        beforeEachSetup();
+        SQLServerDataSource dsLocal = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionString, dsLocal);
+        dsLocal.setSendTemporalDataTypesAsStringForBulkCopy(false);

```

Comment:
this test doesn't actually test the datasource settings as updateDataSource hasn't been updated to include the new properties

Reply:
I think it is testing the datasource, because I'm calling setSendTemporalDataTypesAsStringForBulkCopy instead of using updateDataSource. I just checked as well to make sure it is using the datasource with the connection property set to false.

Reply:
sorry, you're right you are, as you're calling that explicitly after. But that's not how the other connection properties are handled. And updateDataSource needs to be updated to parse all the connection properties properly

Reply:
sure, i'll add that to my tests too then.